drainer/syncer.go

```diff
@@ -323,8 +322,13 @@ func (s *Syncer) sync(executor executor.Executor, jobChan chan *job, executorIdx
 		sqls = sqls[0:0]
 		args = args[0:0]
 		commitTSs = commitTSs[0:0]
-		s.lastSyncTime = time.Now()
-		lastSyncTime = time.Now()
+		now := time.Now()
+		s.lastSyncTime = now
+		if !maxExecutionWaitTimer.Stop() {
+			<-maxExecutionWaitTimer.C
+		}
+		maxExecutionWaitTimer.Reset(maxExecutionWaitTime)
```

Comment:
Is it more straight forward if we reset the timer in the `for` loop right before the `select`?

Reply:
Onely need to reset when `execute` is called